duckduckgo/Android
# Android: Authentication popup shown in the context of previous website

Task/Issue URL: https://app.asana.com/0/414730916066338/1173393687766921/f
Tech Design URL: 
CC: 

**Description**:
When authentication popup is show, previous website is still loaded in the background. This could add some confusion to users if the popup is triggered by an non related website.

Update the URL and hide web content if authentication popup is triggered by a host different to the website currently being displayed. Display content again when dialog dismissed.

**Steps to test this PR**:
> Test 1
1. visit http://lcamtuf.coredump.cx/authspoof/
1. click on the button
1. browser navigates to google.com
1. wait 3 sec for the authentication popup
1. Ensure url gets updated and web content hides

> Test 2
1. visit http://lcamtuf.coredump.cx/authspoof/
1. click on the button
1. browser navigates to google.com
1. wait 3 sec for the authentication popup
1. Ensure url gets updated and web content hides
1. press cancel
1. Ensure web content appears and content is updated

> Test 3
1. visit http://lcamtuf.coredump.cx/authspoof/
1. click on the button
1. browser navigates to google.com
1. wait 3 sec for the authentication popup
1. Ensure url gets updated and web content hides
1. press back or click outside the dialog
1. Ensure web content appears and content is updated

> Test 4
1. visit http://lcamtuf.coredump.cx/authspoof/
1. click on the button
1. browser navigates to google.com
1. wait 3 sec for the authentication popup
1. Ensure url gets updated and web content hides
1. Send the app to background
1. Bring back the app to foreground
1. Ensure web content is visible and updated like user canceled the popup

> Test 5
1. visit http://lcamtuf.coredump.cx/authspoof/
1. click on the button
1. browser navigates to google.com
1. wait 3 sec for the authentication popup
1. Ensure url gets updated and web content hides
1. Send the app to background
1. press sign in
1. Ensure web content appears and content is updated

> Test 6
1. visit http://httpbin.org/basic-auth/test/12345
1. Ensure authentication popup appears
1. introduce wrong credentials (eg.: fake - fake)
1. Ensure authentication popup appears again
1. introduce correct credentials ( test - 12345)
1. Ensure web content is displayed and popup dismissed

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
